src/sentry/replays/query.py

```diff
@@ -274,6 +275,44 @@ def _grouped_unique_scalar_value(
     )
 
 
+def _sorted_aggregated_urls(agg_urls_column, alias):
+    mapped_urls = Function(
+        "arrayMap",
+        parameters=[
+            Lambda(
+                ["url_tuple"], Function("tupleElement", parameters=[Identifier("url_tuple"), 2])
+            ),
+            agg_urls_column,
+        ],
+    )
+    mapped_sequence_ids = Function(
+        "arrayMap",
+        parameters=[
+            Lambda(
+                ["url_tuple"], Function("tupleElement", parameters=[Identifier("url_tuple"), 1])
+            ),
+            agg_urls_column,
+        ],
+    )
```

Comment:
This could be simplified by removing `agg_urls` and using `groupArray` for `urls` and `segment_id` separately.

Reply:
are urls and segment_id guaranteed to be in the same order if they're in separate arrays?

Reply:
seems like we can't guarantee that since segment_id can be null, see here https://github.com/ClickHouse/ClickHouse/discussions/42338#discussioncomment-3887170

Reply:
yes.  grouping applies the functions row by row.

Reply:
see link, don't think it will work in this case.

Reply:
will try a test case for the possible failure mode to confirm.

Reply:
i should have refreshed.  yeah the null condition breaks this.